src/main/java/com/drajer/bsa/service/impl/RrReceiverImpl.java

```diff
@@ -239,18 +241,37 @@ private boolean submitResponseToRestApi(
     logger.info("Cda Rr Model:{}", rrModel);
 
     boolean isSubmitSuccess = false;
-    RestTemplate restTemplate = new RestTemplate();
-    HttpHeaders headers = new HttpHeaders();
-    headers.setContentType(MediaType.APPLICATION_XML);
 
-    HttpEntity<String> request = new HttpEntity<>(rrXml, headers);
+    /**
+     * Only send documents which are reportable to the EHRV8 Extension API
+     */
+    if (phm.getResponseProcessingInstruction() != null) {
+      boolean isReportable = (phm.getResponseProcessingInstruction().equals(EicrTypes.ReportabilityType.RRVS1.toString()) ||
+        phm.getResponseProcessingInstruction().equals(EicrTypes.ReportabilityType.RRVS2.toString()));
 
-    ResponseEntity<?> response =
-        restTemplate.exchange(
-            hs.getHandOffResponseToRestApi(), HttpMethod.POST, request, String.class);
+      if (!isReportable) {
+        logger.info("Reportability Response document isn't reportable.");
+        return true;
+      }
 
-    if (response.getStatusCode().is2xxSuccessful()) {
-      isSubmitSuccess = true;
+      /** Call EHRV8 Extension API to save document */
+      RestTemplate restTemplate = new RestTemplate();
+      HttpHeaders headers = new HttpHeaders();
+      headers.setContentType(MediaType.APPLICATION_JSON);
+
+      int encounterId = Integer.parseInt(rrModel.getEncounterId().getValue().replace("enc-id-", ""));
```

Comment:
Do you think we could check if we have a valid integer after string replacement before integer.parse? I don't know if java has Integer.tryParse thing though.

Reply:
I learned Java doesn't have any built-in tryparse for integers, but that's a good point I can add some integer validation here